src/Internal/Textfield/Implementation.elm

```diff
@@ -284,6 +284,7 @@ textField lift model options _ =
                 [
                   cs "mdc-text-field__input"
                 , css "outline" "none"
+                , Options.attribute (Html.id index)
```

Comment:
Here are two drawbacks that I think should be taken into account with this implementation:
1 - there is not necessarily only one mdc model in an Elm application so the Index could be used several times in a page and we would have several elements with the same id.
2 - without an explicit warning in documentation there are chances that the user creates an element with the same id as the native control.